src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -833,23 +835,32 @@ - (void)test003_ReceiveQueryImageRequestWhileHandlingBDX_RespondImplicitBusy
                                         softwareVersion:kUpdatedSoftwareVersion_5
                                   softwareVersionString:kUpdatedSoftwareVersionString_5
                                              completion:completion];
-        [queryExpectation1 fulfill];
```

Comment:
Why the ordering change here?  If the new order is important, should probably have a comment explaining that it is and why.

Reply:
Well, the expectations refer to receiving delegate callbacks, so it's easier to understand / be sure that the expectation order we assert later via those expectations is correct if fulfilling the expectation is actually the first thing we do in the block. Further below we had

```
[sOTAProviderDelegate respondErrorWithCompletion:outerCompletion];
[queryExpectation2 fulfill];
```

where in response to the query we cancel the existing transfer. Depending how exactly the callbacks are dispatched this could have the effect of the TransferEnd callback being called before we fulfill that expectation.
(With the current way we dispatch them it would probably have been fine, but it's just harder to understand.)

I moved the fulfills to the start of the callbacks as a separate commit first and ensured the tests still pass that way before making any of the other changes.